### PR TITLE
Suggestion popup

### PR DESCRIPTION
**Pull request by emaohn**
- Idea button that triggers popup for suggestions --> animated from bottom
- By default loads saved user preferences (from trip settings) into preference bar on side, but users can adjust and filter as they wish without changing their default preferences
- Display user preference bar on side for users to filter through suggestions on the spot (scrollable)
- Tab controller to display either activity suggestions or food suggestions --> will render Zach's component below the tab bar passing in whichever array is needed.
- Loading animation when querying places from API
- Retrieves suggestions based on user preferences (either passed in or specified) and ensures everything is well formatted to be passed into Zach's component
- Changed trip setting form from popover to modal in order to properly render in center of screen
- Trip functions are setup to be passed down to timeline to retrieve proper parameters for retrieving suggestions (radius, coordinates, time-range) and should be able to integrate with Dan's component through that.

Screen Shot 2020-07-17 at 5 20 49 PM
Screen Shot 2020-07-17 at 5 21 04 PM
Screen Shot 2020-07-17 at 5 23 15 PM

### CODE REVIEW

File: step-capstone/src/styles/SuggestionPopup.css [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    height: 350px;
+}
+
+.scroll {
+    overflow-y: auto;
+}
+
+.test {
+    background-color: aquamarine;
+}
```
Comment (zirving) [inline step-capstone/src/styles/SuggestionPopup.css:16]: I think this is left in accidentally?

Comment (emaohn) [reply step-capstone/src/styles/SuggestionPopup.css:16]: Yep, thanks for catching it!

File: step-capstone/src/components/Pages/Trip.js [hunk 2]
```diff
@@ -33,7 +26,10 @@ export default class Trip extends React.Component {
             },
             map: null,
             service: null,
-            queryResults: null
+            queryResults: null,
+            palceIds: new Set(),
```
Comment (ganewman42) [inline step-capstone/src/components/Pages/Trip.js:30]: Misspelling, should be `placeIds`

File: step-capstone/src/components/Utilities/SuggestionPopup.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        }
+
+        this.handleUserPrefChange = this.handleUserPrefChange.bind(this);
+        this.handleToggleTab = this.handleToggleTab.bind(this);
+        this.renderSuggestions = this.renderSuggestions.bind(this);
+        this.getSuggestions = this.getSuggestions.bind(this);
+    }
+
+    componentDidMount() {
+        // this.getSuggestions();
```
Comment (ganewman42) [inline step-capstone/src/components/Utilities/SuggestionPopup.js:40]: Nit: Please remove or replace with a TODO. If this function should do nothing, there should be some sort of pass you can use.

File: step-capstone/src/components/Utilities/SuggestionPopup.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        if (!this.state.suggestionsLoaded) {
+            return (
+                <Grid container justify="center" align-items="center">
+                    <Box mt={10}>
+                        <CircularProgress color="secondary" />
+                    </Box>
+                </Grid>
+            )
+        }
+        // TODO: Suggestion component
```
Comment (ganewman42) [inline step-capstone/src/components/Utilities/SuggestionPopup.js:120]: Nit: What does this TODO mean? Isn't this the suggestion component?

**Pull request merged**